test/dtypes/test_nf4.py

```diff
@@ -44,55 +45,58 @@ def _build_bnb_linear(input_weight, device):
     bnb_linear.to(device)
     return bnb_linear
 
```

Comment:
* moving TestMod outside since pytest complains about `TestMod.__init__`
* remove `TestCase` to favor `@pytest.mark.parametrize`

Reply:
use `@parametrize` from torch resolved the problem